src/solver/groupnorm/forward_groupnorm.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/groupnorm/solvers.hpp>
+
+#include <miopen/groupnorm/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/groupnorm.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/target_properties.hpp>
+
+#define LOCAL_SIZE 1024
```

Comment:
1024 looks a bit huge.
Is this `LOCAL_SIZE` optimal for all kind of GPUs? Probably it should be a tunable solver.

@atamazov @JehandadKhan what do you think?

Reply:
It's only tested on gfx90a.

Reply:
@kyeonghwanryu @CAHEK7 
> Probably it should be a tunable solver.

Is it important in this context? Does tuning affect required LDS size?

Reply:
It affects required LDS size. (and yes, it's an another case where we can get a benefit of passing actual LDS in runtime instead of recompiling the kernel)
Such workgroup size is not very common across the typical kernels, usually it is significantly smaller. That's why I asked the question - it may be manually tuned on some particular configuration and may not be so efficient on the others.

Reply:
Sorry but I'm not quite familiar with the concept of a tunable solver.
Any suggestion on this?